synology/start_tunnel_zabbix_proxy_to_server.sh

```diff
@@ -0,0 +1,79 @@
+#! /usr/bin/env bash
+server_key="ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIAQzSeGYj0r2P80lGa6qgRGi1OBgJ4q0qvXy+PlZ3BuP"
+sshrelay="sshrelay.ocb.msf.org"
+sshrelay_ip="15.188.17.148,185.199.180.11,2a05:d012:209:9a00:8e2a:9f6c:53be:df41"
+sshrelay_key="ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIDsn2Dvtzm6jJyL9SJY6D1/lRwhFeWR5bQtSSQv6bZYf"
+
+declare -a relay_ports=("443" "22" "80")
+
+zabbix_server_port="${1}"
+priv_key="${2}"
+
+
+if [ -z "${zabbix_server_port}" ]  || [ -z "${priv_key}" ]; then
+  echo "Usage:"
+  echo "  ${0} <zabbix_server_port> <priv_key>"
+  exit 1
+fi
+
+if [ -z "${tmp_dir}" ]; then
+  tmp_dir=$(mktemp -d)
+fi
+
+known_hosts_file="${tmp_dir}/known_hosts"
+
+cat <<EOF > "${known_hosts_file}"
+${localhost}       ${server_key}
+[${localhost}]:7105  ${server_key}
+${sshrelay},${sshrelay_ip} ${sshrelay_key}
+EOF
+
+echo "Waiting for autossh to become available..."
+
+while [ ! -x /usr/local/bin/autossh ]; do
+  sleep 10
+done
+
+echo "Starting tunnel to Server..."
+
+ssh_common_options="-F none \
+                    -o ServerAliveInterval=10 \
+                    -o ServerAliveCountMax=5 \
+                    -o ConnectTimeout=360 \
+                    -o LogLevel=ERROR \
+                    -o AddKeysToAgent=yes"
+                    
+for repeat in $(seq 1 20); do
+  for relay_port in "${relay_ports[@]}"; do
+  
+/usr/local/bin/autossh \
+  -M 0 \
+  -N -T \
+  -L 10052:localhost:10051 \
+  ${ssh_common_options} \
+  -i "${priv_key}" \
+  -o "ExitOnForwardFailure=no" \
+  -o ProxyCommand="ssh -W %h:%p \
+                                  -i "${priv_key}" \
+                                  ${ssh_common_options} \
+                                  -o StrictHostKeyChecking=yes \
+                                  -o UserKnownHostsFile=${known_hosts_file} \
+                                  -p ${relay_port} \
+                                  -l tunneller \
+                                  ${sshrelay}" \
+    -o "ExitOnForwardFailure=yes" \
+#     -o "ServerAliveInterval=10" \
+#     -o "ServerAliveCountMax=5" \
+#     -o "ConnectTimeout=360" \
+#     -o "UpdateHostKeys=yes" \
```

Comment:
These lines can be removed, they will give you errors because bash will not continue your lines like this.